### PR TITLE
Update terser to v5.18.2 and verify optional chaining support

### PR DESCRIPTION
**Pull request by sbc100**
Fixes: #15827

### CODE REVIEW

File: test/optimizer/applyImportAndExportNameChanges2-output.js [hunk 1]
```diff
@@ -275,4 +275,4 @@ WebAssembly.instantiate(Module["wasm"], imports).then(output => {
  dynCall_iiii = asm["p"];
  initRuntime(asm);
  ready();
-});
+}));
```
Comment (kripken) [inline test/optimizer/applyImportAndExportNameChanges2-output.js:278]: Why are these needed?

Comment (sbc100) [reply test/optimizer/applyImportAndExportNameChanges2-output.js:278]: They are not needed, but its seems like acorn is producing them now.. I guess its just a slight behaviour change.

As you can see, no actual code size tests regressed so I guess closure is able just erase these.

If we can we would need to investigate why acorn started doing this.. and see if we can disable it.

Comment (sbc100) [reply test/optimizer/applyImportAndExportNameChanges2-output.js:278]: Do you think we care about these braces?

Comment (kripken) [reply test/optimizer/applyImportAndExportNameChanges2-output.js:278]: Yeah, I guess closure fixes this up. Odd terser slightly regressed here but probably this is related to some bugfix for them...

**Review by kripken — APPROVED** [review]
